crates/ruff_linter/src/rules/ruff/rules/tuple_map_int_version_parsing.rs

```diff
@@ -0,0 +1,161 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{Expr, ExprAttribute, ExprCall, ExprName};
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for calls of the form `tuple(map(int, __version__.split(".")))`.
+///
+/// ## Why is this bad?
+/// `__version__` does not always contain integral-like elements.
+///
+/// ```python
+/// import matplotlib  # 3.9.1.post-1
+///
+/// # ValueError: invalid literal for int() with base 10: 'post1'
+/// tuple(map(int, matplotlib.__version__.split(".")))
+/// ```
+///
+/// See also [PEP 440].
+///
+/// ## Example
+/// ```python
+/// tuple(map(int, matplotlib.__version__.split(".")))
+/// ```
+///
+/// Use instead:
+/// ```python
+/// import packaging.version as version
+///
+/// version.parse(matplotlib.__version__)
+/// ```
+///
+/// [PEP 440]: https://peps.python.org/pep-0440/
+#[violation]
+pub struct TupleMapIntVersionParsing;
+
+impl Violation for TupleMapIntVersionParsing {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "`__version__` may contain non-integral-like elements".to_string()
+    }
+}
+
+/// RUF048
+pub(crate) fn tuple_map_int_version_parsing(checker: &mut Checker, call: &ExprCall) {
+    let semantic = checker.semantic();
+
+    let Some(Expr::Call(child_call)) = tuple_like_call_with_single_argument(semantic, call) else {
+        return;
+    };
```

Comment:
Do we need to check whether the `map()` call is nested inside a `tuple` or a `list` call? I feel like both of these would be victim to the same latent bug:

```py
from foo import __version__
from collections import deque

x = deque(map(int, __version__.split('.')))

for element in map(int, __version__.split('.')):
    print(element)
```

If we just flag all `map(int, __version__.split('.'))` calls, regardless of whether they're nested inside `list()` or `tuple()` calls, the rule is more general and we catch more bugs.